docs/linker-attributes.md

```diff
@@ -0,0 +1,78 @@
+# Linker attributes
+
+As the linker scans the referenced code, it warns about non-understood reflection patterns or other code patterns whose behavior may not be preserved by linking. The following attributes influence the pattern-detection and warning behavior.
+
+## DynamicallyAccessedMembers
+
+Gives extra information about dynamic access patterns which would not otherwise be understood. This can tell the linker to keep additional members that are reflected over, making simple reflection patterns linker-friendly.
+
+TODO: provide examples
+
+The attribute specifies which members of a type are dynamically accessed (e.g. by reflection or interop code). It can be applied to IL locations (method parameters, method return parameters, fields, and properties) that hold a representation of a type, and it indicates via DynamicallyAccessedMemberTypes which members of that type are used dynamically. The linker uses information from these attributes to keep additional members of types, which can make some reflection access patterns linker-safe, eliminating the corresponding warnings.
```

Comment:
Can we include DynamicallyAccessedMemberTypes enum here with documentation as well